managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -252,7 +253,8 @@
     protected volatile State state = null;
 
     private final OrderedScheduler scheduledExecutor;
-    private final OrderedExecutor executor;
+    private final ScheduledExecutorService pinnedScheduledExecutor;
+    private final Executor pinnedExecutor;
```

Comment:
If we are using 2 threads, one with the regular executor (which is more efficient) and the other for the `pinnedScheduledExecutor`, wouldn't that mean that we still have more than 1 thread accessing some of the objects?

Would it make sense to use the generic `scheduledExecutor` (just for deferring purposes) and then jump back into the same `pinnedExecutor`?

Reply:
That's true.

Perhaps a more optimal solution would be to have the capability for scheduling tasks on the pinned scheduler. I don't know why this solution isn't available in the underlying Bookkeeper libraries that are used. The benefit of that is that there isn't an additional thread switch when the scheduled task triggers.

Reply:
> Would it make sense to use the generic scheduledExecutor (just for deferring purposes) and then jump back into the same pinnedExecutor?

@merlimat Do you mean scheduledExecutor.schedule(pinnedExecutor.execute() ...) ?
Seems to be a feasible way right now :)